mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -186,7 +186,7 @@ class BinaryClassificationMetrics @Since("3.0.0") (
       mergeValue = (c: BinaryLabelCounter, labelAndWeight: (Double, Double)) =>
         c += (labelAndWeight._1, labelAndWeight._2),
       mergeCombiners = (c1: BinaryLabelCounter, c2: BinaryLabelCounter) => c1 += c2
-    ).sortByKey(ascending = false)
+    ).sortByKey(ascending = false).cache()
```

Comment:
this RDD `counts` seems used twice at most in this method. I think it not worthwhile to cache it.